mkosi/__init__.py

```diff
@@ -2300,21 +2351,33 @@ def install_type1(
 
 
 def install_uki(context: Context, kver: str, kimg: Path, token: str, partitions: Sequence[Partition]) -> None:
+    bootloader_entry = DEFAULT_BOOTLOADER_ENTRY
+
     roothash_value = ""
     if roothash := finalize_roothash(partitions):
-        roothash_value = f"-{roothash.partition("=")[2]}"
+        roothash_value = roothash.partition("=")[2]
+        bootloader_entry += "-&h"
 
     boot_count = ""
     if (context.root / "etc/kernel/tries").exists():
-        boot_count = f'+{(context.root / "etc/kernel/tries").read_text().strip()}'
+        boot_count = (context.root / "etc/kernel/tries").read_text().strip()
+        bootloader_entry += "+&c"
```

Comment:
Let's not execute this if the option is set. so please conditionalize this on `context.config.bootloader_entry_format`